src/backend/db/database_manager.py

```diff
@@ -850,5 +866,363 @@ def delete_ticket(self, ticket_id, customer_guid):
+                        FROM ticket_comments 
+                        WHERE ticket_id = :ticket_id AND comment_uuid = :comment_uuid AND posted_by = :posted_by
+                    """),
+                    {"ticket_id": ticket_id, "comment_uuid": comment_uuid, "posted_by": posted_by}
+                ).fetchone()
+
+                logger.debug(f"Fetch newly created comment_id result: {result}")
+
+                if result:
+                    comment_id = result[0]
+                    logger.debug(f"Created comment with ID: {comment_id}")
+                else:
+                    raise Exception("Failed to retrieve the newly created comment ID.")
+
+            session.commit()
+            logger.debug(f"Transaction committed successfully for comment ID {comment_id}")
+
+            return {"comment_id": comment_id, "comment": comment}
+
+        except ValueError as e:
+            logger.error(f"Validation error: {e}")
+            session.rollback()
+            raise e
+
+        except SQLAlchemyError as e:
+            logger.error(f"Database error creating comment for customer {customer_guid}: {e}")
+            session.rollback()
+            raise e
+
+        except Exception as e:
+            logger.error(f"Unexpected error while creating comment: {e}")
+            session.rollback()
+            raise e
+
+        finally:
+            logger.debug("Closing database session")
+            session.close()
+
+    def get_comment_by_id(self, comment_id, customer_guid, ticket_id):
+        customer_db_name = self.get_customer_db(customer_guid)
+        session = DatabaseManager._session_factory()
+        try:
+            # Check if the database exists
+            logger.debug(f"Checking if database {customer_db_name} exists")
+            result = session.execute(
+                text("SELECT SCHEMA_NAME FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME = :db_name"),
+                {"db_name": customer_db_name}
+            ).fetchone()
+
+            if not result:
+                raise ValueError(f"Database {customer_db_name} does not exist")
+
+            logger.debug(f"Switching to customer database: {customer_db_name}")
+            session.execute(text(f"USE `{customer_db_name}`"))
+
+            # Retrieve comment details
+            comment_query = '''
+            SELECT comment_id, ticket_id, posted_by, comment, created_at, is_edited
+                FROM ticket_comments
+                WHERE comment_id = :comment_id AND ticket_id = :ticket_id
+            '''
+            comment_result = session.execute(
+                text(comment_query),
+                {"comment_id": comment_id, "ticket_id": ticket_id},
+            ).fetchone()
+
+            if not comment_result:
+                return None  # If comment is not found, return None
+
+            # Convert comment result into a dictionary
+            logger.info(f"Comment details by comment_id {comment_result}")
+            comment_data = {
```

Comment:
The output should have

CREATE TABLE IF NOT EXISTS ticket_comments (
comment_id BIGINT AUTO_INCREMENT PRIMARY KEY,
ticket_id BIGINT NOT NULL,
posted_by VARCHAR(255) NOT NULL,
comment TEXT NOT NULL,
is_edited BOOLEAN DEFAULT FALSE,
created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP
);

Reply:
modified sir